### PR TITLE
feat(onPremise): Add email on premise recollection

### PR DESCRIPTION
**Pull request by rsanchezbalo**

### CODE REVIEW

File: app/modules/projects/create/create-project.controller.coffee [hunk 1]
```diff
@@ -12,18 +12,35 @@ class CreateProjectController
...
-        "$tgAuth"
+        "$tgAuth",
+        "tgLightboxFactory",
+        "$tgStorage",
+        "tgResources",
+        "$tgConfig"
     ]
 
-    constructor: (@appMetaService, @translate, @projectService, @location, @authService) ->
+    constructor: (@appMetaService, @translate, @projectService, @location, @authService, @lightboxFactory, @storageService, @rs, @config) ->
```
Comment (Xaviju) [inline app/modules/projects/create/create-project.controller.coffee:22]: Review dependencies, do you use them?

File: app/modules/projects/create/newsletter-email-lightbox/newsletter-email-lightbox.directive.coffee [hunk 2]
```diff
@@ -0,0 +1,59 @@
+###
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# Copyright (c) 2021-present Kaleidos Ventures SL
+###
+
+NewsletterEmailLightboxDirective = (lightboxService, lightboxKeyboardNavigationService, $storageService, $rs, $auth, $currentUserService, $rs2) ->
```
Comment (Xaviju) [inline app/modules/projects/create/newsletter-email-lightbox/newsletter-email-lightbox.directive.coffee:9]: Review dependencies

File: app/modules/projects/create/newsletter-email-lightbox/newsletter-email-lightbox.jade [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    .option-container
+        .ask
+            .custom-checkbox
+                input(
+                    type="checkbox"
+                    name="ask"
+                    id="dont-ask"
+                    ng-model="dontAsk"
+                )
+                label(
```
Comment (Xaviju) [inline app/modules/projects/create/newsletter-email-lightbox/newsletter-email-lightbox.jade:42]: Why the label is empty?

File: app/modules/projects/create/newsletter-email-lightbox/newsletter-email-lightbox.scss [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    &.setting {
+        .lightbox-title {
+            @include font-size(small);
+
+            text-align: left;
+        }
+    }
+
+    form {
+        flex-basis: 700px;
```
Comment (Xaviju) [inline app/modules/projects/create/newsletter-email-lightbox/newsletter-email-lightbox.scss:12]: Is this a child of a flex item?

File: app/coffee/modules/user-settings/notifications.coffee [hunk 5]
```diff
@@ -29,15 +29,39 @@ class UserNotificationsController extends mixOf(taiga.Controller, taiga.PageMixi
...
-        "tgErrorHandlingService"
+        "tgErrorHandlingService",
+        "tgProjectService",
+        "$tgConfig",
+        "tgResources",
+        "tgCurrentUserService"
     ]
 
-    constructor: (@scope, @rootscope, @repo, @confirm, @rs, @params, @q, @location, @navUrls, @auth, @errorHandlingService) ->
+    constructor: (@scope, @rootscope, @repo, @confirm, @rs, @params, @q, @location, @navUrls, @auth, @errorHandlingService, @storageService, @config, @resources, @currentUserService) ->
```
[the remark this answers is not in the document]
Comment (rsanchezbalo) [reply app/coffee/modules/user-settings/notifications.coffee:39]: No, you are totally right. There are others that are not being used.

File: app/modules/projects/create/newsletter-email-lightbox/newsletter-email-lightbox.scss (hunk 4 above)
Comment (rsanchezbalo) [reply app/modules/projects/create/newsletter-email-lightbox/newsletter-email-lightbox.scss:12]: Yeah, the CSS here is a little weird, just changed something, so now this is not necessary

**Pull request merged**